src/NServiceBus.Metrics/MetricsOptions.cs

```diff
@@ -65,14 +65,16 @@ public void EnableCustomReport(Func<string, Task> func, TimeSpan interval)
         /// </summary>
         /// <param name="serviceControlMetricsAddress">The transport address of the ServiceControl instance</param>
         /// <param name="interval">Interval between consecutive reports</param>
+        /// <param name="instanceId">Unique identifier of endpoint physical instance</param>
         [ObsoleteEx(Message = "Not for public use.")]
-        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval)
+        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval, string instanceId)
```

Comment:
I think that the biggest problem is Azure. For Azure `HostId`=`RoleId` and `DisplayName`=`RoleName`. So the situation we end up with is that for non Azure endpoints we should show `DispalyName` and for Azure we should display `HostId` or `DisplayName+HostId`. 

I think that problem of hostId and displayname should be tackled separately. Adding parameter to the API is not ideal but it enables us to handle all environments in the same way.